### PR TITLE
feat: Add landing page list type

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-2609

## Proposed changes

- Adds LandingPage list type
- Adds `LandingPage` to Article category enum
- Adds custom view to display tagged articles for landing page
- Tags are used to associate articles with the landing page.
  - The landing page itself can have one tag, set in the `Article Tag` field.
  - An article can have _many_ tags. If it is tagged with the same tag that is in the `Article Tag` field, it will be displayed on the landing page.
  - On the Landing Page in the CMS, articles that will be displayed are listed out under the `Articles` field. This is a read-only field.
  - **TODO:** 
    - ~Build out this view a bit more -- link to view on portal, along with link to edit in CMS (current link)~
    - ~Add fields to virtual field to match the data we'll need on the front-end to display articles on landing page~
  
Screenshot 2023-10-31 at 5 00 53 PM
Screenshot 2023-10-31 at 5 01 26 PM

  
  
  
  
## Reviewer notes
Related PR: https://github.com/USSF-ORBIT/ussf-portal-client/pull/1143
E2E test (under a different branch name): https://github.com/USSF-ORBIT/ussf-portal/pull/332

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-client
yarn dev
```

Login to the cms http://localhost:3001

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

File: src/components/landingPageArticles.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+
+import {
+  FieldController,
```
Comment (abbyoung) [inline src/components/landingPageArticles.tsx:4]: todo -- clean up

**Review by gidjin — COMMENTED** [review]
LGTM, I'm wondering about access for Managers to edit landing pages.

File: src/lists/LandingPage.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+const SLUG_REGEX = /^[a-z0-9]+(?:-[a-z0-9]+)*$/
+const SLUG_MAX = 1000
+
+const LandingPage = list(
+  withTracking({
+    access: {
+      operation: {
+        create: isAdmin,
+        query: () => true,
+        update: isAdmin,
```
[the remark this answers is not in the document]
Comment (abbyoung) [reply src/lists/LandingPage.ts:22]: let's check with @jbecker01 -- i thought for v1 it was admin only but i'm happy to update it

File: src/lists/LandingPage.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+              } else if (slug.length > SLUG_MAX) {
+                addValidationError(
+                  `Slug is too long (maximum of ${SLUG_MAX} characters)`
+                )
+              } else if (!SLUG_REGEX.test(slug)) {
+                addValidationError(
+                  'Slug must be a valid slug (only alphanumeric characters and dashes allowed)'
+                )
+              }
+            }
```
[the remark this answers is not in the document]
Comment (abbyoung) [reply src/lists/LandingPage.ts:89]: The Page Title field is marked as required, so it'll throw a validation error if it's left blank.

File: src/lists/LandingPage.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gidjin) [reply src/lists/LandingPage.ts:22]: As discussed this is a separate story: https://app.shortcut.com/orbit-truss/story/2705/allow-cms-managers-to-update-landing-pages

**Review by gidjin — APPROVED** [review]

**Pull request merged**